src/lib/form/lib.ts

```diff
@@ -0,0 +1,28 @@
+import { FormManagerFactory } from "../handlePageRequest";
+import { FormJSON } from "./formManager";
+
+export function isValid<T>(form: T, rules: FormManagerFactory): boolean {
+  const formManager = rules(form);
+
+  formManager.markAsDirty();
+
+  return formManager.isValid();
+}
+
+export function withErrorMessages<T>(
+  form: T,
+  rules: FormManagerFactory
+): FormJSON {
+  const formManager = rules(form);
+
+  formManager.markAsDirty();
+
+  return formManager.serialise();
+}
+
+export function withoutErrorMessages<T>(
+  form: T,
+  rules: FormManagerFactory
+): FormJSON {
+  return rules(form).serialise();
+}
```

Comment:
Helper functions to hide the concept of a form being dirty or not.  Used by the `Rule`s to conditionally display error messages.